### PR TITLE
Make Ice.Current non-nullable in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR updates the generated code and Ice.Object to make the Ice.Current parameter non-nullable (and without a default).

It also enabled nullable in Object.cs.

### CODE REVIEW

File: csharp/src/Ice/Object.cs [hunk 1]
```diff
@@ -192,20 +194,20 @@ public static string ice_staticId()
     };
 
     /// <summary>
-    /// Dispatches an invocation to a servant. This method is used by dispatch interceptors to forward an invocation
+    /// Dispatches an dispatch to a servant. This method is used by dispatch interceptors to forward an dispatch
```
Comment (pepone) [inline csharp/src/Ice/Object.cs:197]:
```suggestion
    /// Dispatches a dispatch to a servant. This method is used by dispatch interceptors to forward a dispatch
```

File: csharp/src/Ice/Object.cs [hunk 2]
```diff
@@ -27,40 +29,40 @@ public interface Object : ICloneable
...
     /// </summary>
-    /// <param name="current">The Current object for the invocation.</param>
+    /// <param name="current">The Current object for the dispatch.</param>
     /// <returns>The Slice type ID of the most-derived interface.</returns>
-    string ice_id(Current current = null);
+    string ice_id(Current current);
 
     /// <summary>
-    /// Dispatches an invocation to a servant. This method is used by dispatch interceptors to forward an invocation
+    /// Dispatches an dispatch to a servant. This method is used by dispatch interceptors to forward an dispatch
```
Comment (pepone) [inline csharp/src/Ice/Object.cs:58]:
```suggestion
    /// Dispatches a dispatch to a servant. This method is used by dispatch interceptors to forward a dispatch
```

**Review by pepone — APPROVED** [review]

File: csharp/src/Ice/Object.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/src/Ice/Object.cs]: Fixed.

File: csharp/src/Ice/Object.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/src/Ice/Object.cs]: Fixed

File: csharp/src/Ice/Object.cs [hunk 3]
```diff
@@ -27,40 +29,40 @@ public interface Object : ICloneable
...
     /// </summary>
-    /// <param name="current">The Current object for the invocation.</param>
+    /// <param name="current">The Current object for the dispatch.</param>
     /// <returns>The Slice type ID of the most-derived interface.</returns>
-    string ice_id(Current current = null);
+    string ice_id(Current current);
 
     /// <summary>
-    /// Dispatches an invocation to a servant. This method is used by dispatch interceptors to forward an invocation
+    /// Dispatches a dispatch to a servant. This method is used by dispatch interceptors to forward a dispatch
```
Comment (externl) [inline csharp/src/Ice/Object.cs:58]: I don't really like "Dispatches a dispatch", there's no type to match this and it sounds odd.

The type is a request, I think it makes sense to just say that.
```suggestion
    /// Dispatches a request to a servant. This method is used by dispatch interceptors to forward a dispatch
```

**Review by externl — APPROVED** [review]
Looks good. See comments.

**Pull request merged**